src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -2589,6 +2566,11 @@ MavlinkReceiver::receive_thread(void *arg)
 	hrt_abstime last_send_update = 0;
 
 	while (!_mavlink->_task_should_exit) {
+		// Check for updated parameters.
+		if (_param_update_sub.updated()) {
```

Comment:
@dagar and @davids5 , this check mirrors the `Navigator::run()` method while loop check for updated parameters.  This should prevent the issue discussed during the dev call checking for updated params every iteration of the `receive_thread()` while loop.